nilearn/image/tests/test_image.py

```diff
@@ -938,6 +996,40 @@ def test_binarize_negative_img(img_4d_rand_eye):
     assert_array_equal(np.unique(img_original.dataobj), np.array([0, 1]))
 
 
+def test_binarize_img_copied_header(img_4d_mni_tr2):
+    # Test equality of header fields between input and output
+    result = binarize_img(img_4d_mni_tr2, threshold=0.5, copy_header=True)
+    # only the min value should be different
+    match_headers_keys(
+        result,
+        img_4d_mni_tr2,
+        except_keys=["cal_min", "cal_max"],
+    )
+    # min value should be 0 in the result
+    assert result.header["cal_min"] == 0
+    # max value should be 1 in the result
+    assert result.header["cal_max"] == 1
+
+
+@pytest.mark.parametrize(
```

Comment:
@Remi-Gau, here I test the warning being thrown for `copy_header=False` (for all `image.py` functions)

Reply:
Just so you dont have to review everything again